Ubuntu: handle changes in os-prober 1.33
Upstream: backport of r2594
Description: Cope with Windows 7 in os-prober output.

diff -Nur -x '*.orig' -x '*~' grub2-1.97~beta3/util/grub.d/30_os-prober.in grub2-1.97~beta3.new/util/grub.d/30_os-prober.in
--- grub2-1.97~beta3/util/grub.d/30_os-prober.in	2009-09-15 06:45:34.000000000 +0100
+++ grub2-1.97~beta3.new/util/grub.d/30_os-prober.in	2009-09-15 06:46:42.000000000 +0100
@@ -102,11 +102,16 @@
       save_default_entry | sed -e "s/^/\t/"
       prepare_grub_to_access_device ${DEVICE} | sed -e "s/^/\t/"
 
-      if [ "${LONGNAME}" != "Windows Vista (loader)" ] ; then
-        cat << EOF
+      case ${LONGNAME} in
+	Windows\ Vista*|Windows\ 7*)
+	;;
+	*)
+	  cat << EOF
 	drivemap -s (hd0) \${root}
 EOF
-      fi
+	;;
+      esac
+
       cat <<EOF
 	chainloader +1
 }
